### PR TITLE
replace update in setupmaster.go

### PR DESCRIPTION
**Pull request by xilabao**
follow the TODO

### CODE REVIEW

**Comment by luxas** [comment]
@xilabao Thanks for this contribution!! This is great!

cc @deads2k

@k8s-bot ok to test

File: cmd/kubeadm/app/phases/apiconfig/setupmaster.go [hunk 1]
```diff
@@ -51,15 +53,16 @@ func attemptToUpdateMasterRoleLabelsAndTaints(client *clientset.Clientset) error
...
-	n.ObjectMeta.Labels[metav1.NodeLabelKubeadmAlphaRole] = metav1.NodeLabelRoleMaster
-
 	// TODO: Use the Taints beta field on the NodeSpec now
 	taintsAnnotation, _ := json.Marshal([]v1.Taint{{Key: "dedicated", Value: "master", Effect: "NoSchedule"}})
-	n.ObjectMeta.Annotations[v1.TaintsAnnotationKey] = string(taintsAnnotation)
 
-	// TODO: Use a patch instead of an Update
-	if _, err := client.Nodes().Update(n); err != nil {
+	// TODO: Switch to the new master label defined in https://github.com/kubernetes/kubernetes/pull/39112
+	addAnnotationPatch := fmt.Sprintf(
```
Comment (deads2k) [inline cmd/kubeadm/app/phases/apiconfig/setupmaster.go]: I wouldn't recommend building a patch by hand.  As I recall there are functions that take two objects and compute the diff.

**Comment by xilabao** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by xilabao** [comment]
@deads2k @luxas PTAL

**Comment by pires** [comment]
ah, I was assigned but then removed :/

**Comment by luxas** [comment]
@pires Ah, yeah, sorry. Discussed this on Slack yesterday, where you said you had the rest of the week full.

Anyway, this lgtm as well
/lgtm

**Pull request merged**